### PR TITLE
Add bzip2 to known mimetypes

### PR DESCRIPTION
**Pull request by VicDeo**
Ref https://github.com/owncloud/core/issues/18880#issuecomment-160769693

And fix it's support by `OC_Archive`
see https://github.com/owncloud/core/blob/master/lib/private/archive.php#L46

### CODE REVIEW

**Comment by VicDeo** [comment]
@MorrisJobke @PVince81

**Comment by MorrisJobke** [comment]
@karlitschek Another update performance killer :( I guess we need to rethink all of this

**Comment by VicDeo** [comment]
@MorrisJobke I haven't checked internals. 
Shouldn't it be done with a single `UPDATE table set mime='alice' where mime='bob'` query?

**Comment by oparoz** [comment]
I agree with @MorrisJobke, media types can be repaired in the background, it's not that important to have it all working seconds after having updated, especially if there is a way to track the progress of background jobs.

**Comment by karlitschek** [comment]
Exactly as @oparoz said. This should happen later in a background job.
We also have to think VERY carefully which mime-types we really need here. There is no need for a complete list of all existing mimetypes. We only need the ones where we have thumbnails or apps that want to select files based on mimetypes like music or photos. All other mimetypes don't matter for ownCloud.

**Comment by MorrisJobke** [comment]
> And fix it's support by OC_Archive
> see https://github.com/owncloud/core/blob/master/lib/private/archive.php#L46

It seems to fix an updater issue. Mmmmh ... tricky one. @VicDeo what is the timespan this needs to be fixed? If this could also wait for 9.1 I guess we should rethink the whole mime type update scenario here and maybe create a one time background job for this.

**Comment by VicDeo** [comment]
@MorrisJobke this is not that critical unless we will switch all channels to `bzip` again. Currently only  daily channel uses `bzip` while other channels provide `zip` packages.
New updater is not affected, it doesn't rely on `OC_Archive`.
It just make impossible to use daily channel for all oC _before_ 9.0

**Comment by LukasReschke** [comment]
I'd advocate to revert the automatic correction of the mimetype in the repair step. Then all new files would have the correct mimetype while old files just keep the current one.

Shouldn't be too bad…

**Comment by rullzer** [comment]
I vote to just have an occ command that does this. So we can advice admins to run that command at a later date.

**Comment by karlitschek** [comment]
occ commands should be for optional admin actions only. if we think this is part of a full update then we should make sure that it is executed. obviously asyncroniously. maybe as a background task

**Comment by PVince81** [comment]
How about splitting this PR in two parts, then ?

One part just adds the new mimetype which is needed for the updater in some way, and the second one will take care of repairing, using whichever method is most suitable (bkg job I guess)

**Comment by DeepDiver1975** [comment]
This repair step is a classical background job repair step - to be implemented in the scope of 9.1 - on it ...

**Comment by MorrisJobke** [comment]
> One part just adds the new mimetype which is needed for the updater in some way, and the second one will take care of repairing, using whichever method is most suitable (bkg job I guess)

@VicDeo Could you please remove the repair statement then and @DeepDiver1975 will work on it in a separate PR? Then we could at least move this forward.

**Comment by DeepDiver1975** [comment]
> @VicDeo Could you please remove the repair statement then and @DeepDiver1975 will work on it in a separate PR? Then we could at least move this forward.

we can keep this in here alltogether - I'll simply move the repair step later then

**Comment by DeepDiver1975** [comment]
@VicDeo this is conflicting ...

**Comment by VicDeo** [comment]
@DeepDiver1975 rebased

**Comment by PVince81** [comment]
Rebased due to version.php conflict.

Uploading bz2 files shows the correct mime type.
I'd expect the icon to be a package instead:

bz2-icon

Anyway, the purpose of this PR is to bring the mimetype string to fix some updater problems.

:+1: fine by me

**Comment by PVince81** [comment]
Raised https://github.com/owncloud/core/issues/24629 for the background job

**Comment by MorrisJobke** [comment]
I would like to remove the repair steps, because they will produce trouble.

File: lib/private/Repair/RepairMimeTypes.php [hunk 1]
```diff
@@ -375,5 +385,11 @@ public function run(IOutput $out) {
 				$out->info('Fixed richdocuments additional office mime types');
 			}
 		}
+		
+		if (version_compare($ocVersionFromBeforeUpdate, '9.1.0.1', '<')) {
```
Comment (nickvergessen) [inline lib/private/Repair/RepairMimeTypes.php]: needs to be updated

Comment (MorrisJobke) [inline lib/private/Repair/RepairMimeTypes.php]: I would simply remove the repair step and only add the mimetype (repair is part of https://github.com/owncloud/core/issues/24629)

Comment (PVince81) [inline lib/private/Repair/RepairMimeTypes.php]: @VicDeo

**Comment by VicDeo** [comment]
@PVince81 @DeepDiver1975 @rullzer 
rebased, conflicts are solved, repair steps  are removed as requested

File: version.php [hunk 2]
```diff
@@ -25,7 +25,7 @@
 // We only can count up. The 4. digit is only for the internal patchlevel to trigger DB upgrades
 // between betas, final and RCs. This is _not_ the public version number. Reset minor/patchlevel
 // when updating major/minor version number.
-$OC_Version = array(9, 1, 0, 5);
+$OC_Version = array(9, 1, 0, 6);
```
Comment (PVince81) [inline version.php]: not sure if this is still needed, but ok

**Comment by PVince81** [comment]
:+1:

File: resources/config/mimetypemapping.dist.json [hunk 3]
```diff
@@ -19,6 +19,7 @@
 	"bin": ["application/x-bin"],
 	"bmp": ["image/bmp"],
 	"bpg": ["image/bpg"],
+	"bz2" : ["application/x-bzip2"],
```
Comment (nickvergessen) [inline resources/config/mimetypemapping.dist.json]: how strong is json with layout? otherwise the space before `:` could cause issues?

Comment (VicDeo) [inline resources/config/mimetypemapping.dist.json]: @nickvergessen it shoulnd't but I'll remove it anyway to have the things consistent. Thanks!

Comment (VicDeo) [inline resources/config/mimetypemapping.dist.json]: @nickvergessen removed

**Comment by nickvergessen** [comment]
:+1:

**Pull request merged**